src/app/mainwinqt5.cpp

```diff
@@ -0,0 +1,215 @@
+/***************************************************************************
+    mainwin.cpp
+    ---------------------
+    begin                : February 2017
+    copyright            : (C) 2017 by Juergen E. Fischer
+    email                : jef at norbit dot de
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include <windows.h>
+#include <io.h>
+
+#include <sstream>
+#include <iostream>
+#include <fstream>
+#include <list>
+#include <memory>
+
+void showError( std::string message, std::string title )
```

Comment:
Do we need a specific Qt5 version of mainwin.cpp because Qt5 is somehow is incompatible with wchar?

In this case, wouldn't be better to go with macro to switch between char/wchar methods/types ?

Reply:
I was exploring a backwards compatible way in https://github.com/qgis/QGIS/pull/56841#issuecomment-2024650570 but had to refrain because I didn't get a response.
I am aware that this duplication approach is not DRY, but it's for a transition period only, so it's probably better to stick to a solution that works with Qt5 and prepare for a unicode future in parallel.

Reply:
I agree, that's non blocking